envs/development/conda/fedbiomed-researcher-macosx-arm.yaml

```diff
@@ -0,0 +1,74 @@
+#
+# environment for fedbiomed-researcher on macosx
+#
+#
+#
+name: fedbiomed-researcher
+
+channels:
+  - conda-forge
+
+dependencies:
+  # minimal environment
+  - python >=3.10,<3.11
+  - pip >= 23.0
+  # TODO: consider migrating for classical notebooks to notebook7 or jupyterlab
+  # https://jupyter-notebook.readthedocs.io/en/latest/migrate_to_notebook7.html
+  - notebook ~=6.5.0
+  - jupyter ~=1.0.0
+  - ipython ~=8.13.2
+  # tests
+  - pytest ~=7.2.0
+  - pytest-cov ~=4.1.0
+  - tinydb ~=4.7.1
+  - tabulate >=0.9.0,<0.10.0
+  # tools
+  - colorama >=0.4.6,<0.5
+  # code
+  - psutil ~=5.9.0
+  - requests ~=2.29.0
+  - validators >=0.20.0,<0.21.0
+  - git ~=2.40.1
+  - packaging ~=23.1
+  # these two have to be aligned
+  - cryptography ~=40.0.0
+  - pyopenssl ~=23.1.1
+  # git notebook striper - TODO: consider removing if not used anymore ...
+  - nbstripout >=0.6.1,<0.7.0
+  - joblib >=1.2.0,<1.3.0
+  # other
+  - pip:
+      # no itk available from conda for macosx-m1
+      - itk >=5.3.0,<5.4.0
+      # sklearn
+      #   + scipy >= 1.9 from conda-forge needs recent GLIBC thus causes issue 389 with many current systems
+      #   + another option is to install scipy from pip which supports older GLIBC
+      - scipy >=1.10.0,<1.11.0
+      - scikit-learn >=1.2.0,<1.3.0
+      # nn
+      # torch 2.x recently released (2023-03) and not yet supported by some deps including: opacus, declearn
+      - torch ~=1.13.0
+      - torchvision >=0.14.0,<0.15.0
+      - opacus >=1.4.0,<1.5.0
+      - monai >=1.1.0,<1.2.0
+      # other
+      - msgpack ~=1.0
+      - persist-queue >=0.5.1,<0.6.0
+      # pandas 2.x recently released (2023-04) but few breaking changes
+      - pandas ~=2.0.1
+      - tensorboard ~=2.13.0
+      - python-minifier ~=2.5.0
+      # for nbconvert
+      - jupyter-contrib-nbextensions >=0.7.0,<0.8.0
+      - pathvalidate ~=3.0.0
+      # FLamby
+      - git+https://github.com/owkin/FLamby@main
+      # declearn
+      - declearn[torch1] ~=2.3.2
+      - gmpy2 >=2.1,< 2.2
+      # grpc
+      - grpcio
```

Comment:
Thank you @j-l-s 

Which version did you have to use for MacOS w/ ARM ?

It would be nice to:
- keep major/minor version specification in the yaml file (to have a future-safe yaml file, when `grpcio` versions on conda-forge evolve)
- if possible: test this version on MacOS Intel + Linux to see if we can use the same `grpcio` major/minor for all platforms

Reply:
I released the versioning in the yaml file, and let conda/mamba deal with the dependancies. Here are the installed version:
```
$ conda list | grep grpc
grpcio                    1.60.0                   pypi_0    pypi
grpcio-tools              1.60.0                   pypi_0    pypi
```